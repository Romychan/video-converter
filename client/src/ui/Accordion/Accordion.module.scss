.accordion {
  &.active .body {
    overflow: unset;
  }
}

.header {
  display: flex;
  align-items: center;
  width: 100%;
  cursor: pointer;
  transition: 0.2s;
  padding: 3px 0px;

  &:hover {
    & .square {
      background: var(--color-secondary-hover);
    }
  }
}

.square {
  width: 30px;
  height: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
  background: var(--color-secondary);
  transition: 0.2s;
  border-radius: 10px;
  margin-right: 10px;
}

.icon {
  stroke-width: 1.2px;
  stroke: var(--text-color-primary);
}

.title {
  font-family: var(--text-family-medium);
  font-size: 0.875rem;
  color: var(--text-color-primary);
  letter-spacing: 0.01rem;
  transition: 0.2s;
}

.body {
  max-height: 0;
  overflow: hidden;
  transition: 0.4s;
}

.content {
  padding: 1.5rem 0px 0.75rem 0px;
}
