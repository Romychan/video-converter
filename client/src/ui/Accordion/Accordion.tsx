import { ReactNode, useState } from 'react';
import clsx from 'clsx';

import { Icon } from '../Icon/Icon';

import styles from './Accordion.module.scss';

interface IAccordionProps {
  title: string;
  children?: ReactNode;
  className?: string;
}

export const Accordion = ({
  title,
  children,
  className = '',
}: IAccordionProps) => {
  const [clicked, setClicked] = useState(false);

  return (
    <div
      className={clsx(styles.accordion, className, {
        [styles.active]: clicked,
      })}
      data-testid="accordion"
    >
      <div
        className={styles.header}
        onClick={() => setClicked(!clicked)}
        data-testid="accordion-header"
      >
        <div className={styles.square}>
          {clicked ? (
            <Icon name="minus" size={14} className={styles.icon} />
          ) : (
            <Icon name="plus" size={14} className={styles.icon} />
          )}
        </div>
        <h3 className={styles.title}>{title}</h3>
      </div>

      <div
        className={styles.body}
        style={clicked ? { maxHeight: '1200px' } : { maxHeight: '0px' }}
        data-testid="accordion-body"
      >
        <div className={styles.content}>{children}</div>
      </div>
    </div>
  );
};
