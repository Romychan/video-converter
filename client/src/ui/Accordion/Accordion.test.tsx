import {
  RenderResult,
  fireEvent,
  render,
  screen,
} from '@testing-library/react';

import { Accordion } from './Accordion';

describe('Accordion', () => {
  let container: RenderResult;

  beforeEach(() => {
    container = render(<Accordion title="Testing">Content</Accordion>);
  });

  it('should render correctly', () => {
    expect(screen.getByText(/Testing/i)).toBeInTheDocument();
  });

  it('should opens and closes', () => {
    const accordion = screen.getByTestId('accordion');
    const header = screen.getByTestId('accordion-header');
    const body = screen.getByTestId('accordion-body');

    fireEvent.click(header);
    expect(body).toHaveStyle({
      'max-height': '1200px',
    });
    expect(accordion.getAttribute('class')).toMatch(/active/gi);

    fireEvent.click(header);
    expect(body).toHaveStyle({
      'max-height': '0px',
    });
    expect(accordion.getAttribute('class')).not.toMatch(/active/gi);
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
