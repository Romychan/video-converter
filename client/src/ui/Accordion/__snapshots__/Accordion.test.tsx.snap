// Vitest Snapshot v1

exports[`Accordion > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_accordion_7e54ec"
        data-testid="accordion"
      >
        <div
          class="_header_7e54ec"
          data-testid="accordion-header"
        >
          <div
            class="_square_7e54ec"
          >
            <svg
              class="icon icon-plus _icon_7e54ec"
              data-testid="icon"
              height="14"
              stroke=""
              width="14"
            >
              <use
                xlink:href="/src/assets/img/general/icons.svg#plus"
              />
            </svg>
          </div>
          <h3
            class="_title_7e54ec"
          >
            Testing
          </h3>
        </div>
        <div
          class="_body_7e54ec"
          data-testid="accordion-body"
          style="max-height: 0px;"
        >
          <div
            class="_content_7e54ec"
          >
            Content
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_accordion_7e54ec"
      data-testid="accordion"
    >
      <div
        class="_header_7e54ec"
        data-testid="accordion-header"
      >
        <div
          class="_square_7e54ec"
        >
          <svg
            class="icon icon-plus _icon_7e54ec"
            data-testid="icon"
            height="14"
            stroke=""
            width="14"
          >
            <use
              xlink:href="/src/assets/img/general/icons.svg#plus"
            />
          </svg>
        </div>
        <h3
          class="_title_7e54ec"
        >
          Testing
        </h3>
      </div>
      <div
        class="_body_7e54ec"
        data-testid="accordion-body"
        style="max-height: 0px;"
      >
        <div
          class="_content_7e54ec"
        >
          Content
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
